projects/igniteui-angular/src/lib/directives/for-of/base.helper.component.ts

```diff
@@ -34,6 +36,9 @@ export class VirtualHelperBaseDirective implements OnDestroy, AfterViewInit {
         protected _zone: NgZone,
         @Inject(DOCUMENT) public document: any,
         protected platformUtil: PlatformUtil,
+        private renderer: Renderer2,
+        @Inject(PLATFORM_ID) private platformId: Object,
+        private ngZone: NgZone
```

Comment:
Try to use the [inject API](https://angular.dev/api/core/inject) so you don't have to also modify all classes that extend the base class.